docker-java-transport-httpclient5/src/main/java/com/github/dockerjava/httpclient5/NamedPipeSocket.java

```diff
@@ -8,44 +8,54 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
-import java.io.RandomAccessFile;
 import java.net.Socket;
 import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousByteChannel;
+import java.nio.channels.AsynchronousCloseException;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.channels.Channels;
+import java.nio.channels.CompletionHandler;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
 
 class NamedPipeSocket extends Socket {
 
     private final String socketFileName;
 
-    private RandomAccessFile file;
-
-    private InputStream is;
-
-    private OutputStream os;
+    private AsynchronousFileByteChannel channel;
 
     NamedPipeSocket(String socketFileName) {
         this.socketFileName = socketFileName;
     }
 
     @Override
     public void close() throws IOException {
-        if (file != null) {
-            file.close();
-            file = null;
+        if (channel != null) {
+            channel.close();
         }
     }
 
     @Override
-    public void connect(SocketAddress endpoint) {
+    public void connect(SocketAddress endpoint) throws IOException {
         connect(endpoint, 0);
     }
 
     @Override
-    public void connect(SocketAddress endpoint, int timeout) {
+    public void connect(SocketAddress endpoint, int timeout) throws IOException {
         long startedAt = System.currentTimeMillis();
         timeout = Math.max(timeout, 10_000);
         while (true) {
             try {
-                file = new RandomAccessFile(socketFileName, "rw");
+                channel = new AsynchronousFileByteChannel(
+                    AsynchronousFileChannel.open(
+                        Paths.get(socketFileName),
+                        StandardOpenOption.READ,
+                        StandardOpenOption.WRITE
+                    )
+                );
                 break;
             } catch (FileNotFoundException e) {
```

Comment:
I wonder if that should catch `NoSuchFileException` now.

Reply:
Nice catch, thanks! 👍